cabal.project

```diff
@@ -2,3 +2,63 @@ packages: .
 
 -- allow-newer: bytestring
 -- constraints: bytestring ^>=0.11
+
+-- TODO: Remove the below vendored packages when they have migrated to GHC 9.2
+
+source-repository-package
```

Comment:
Why add these? What's wrong with [`head.hackage`](https://ghc.gitlab.haskell.org/head.hackage/) versions?

I won't depend on these non-official upstreams. Please revert `cabal.project` changes.

In general, I also won't merge this (nor `aeson`) patch until GHC-9.2 got at least the first release candidate. IMO it's too early to push changes into the ecosystem for it.